src/bijectors/ordered.jl

```diff
@@ -88,3 +78,72 @@ end
 
 logabsdetjac(b::OrderedBijector, x::AbstractVector) = sum(@view(x[2:end]))
 logabsdetjac(b::OrderedBijector, x::AbstractMatrix) = vec(sum(@view(x[2:end, :]); dims=1))
+
+# Need a custom distribution type to handle this properly.
+struct OrderedDistribution{B,D<:ContinuousMultivariateDistribution} <:
+       ContinuousMultivariateDistribution
+    dist::D
+    bijector::B
+end
+
+"""
+    ordered(d::Distribution)
+
+Return a `Distribution` whose support are ordered vectors, i.e., vectors with increasingly ordered elements.
+
+This transformation is currently only supported for otherwise unconstrained distributions.
+"""
+function ordered(d::ContinuousMultivariateDistribution)
+    # We're good if the map from unconstrained (in which we apply the ordered bijector)
+    # to constrained is monotonically increasing, i.e. order-preserving. In that case,
+    # we can form the ordered transformation as `binv ∘ OrderedBijector() ∘ b`.
+    # Similarly, if we're working with monotonically decreasing maps, we can do the same
+    # but with the addition of a sign flip before and after the ordered bijector.
+    b = bijector(d)
+    binv = inverse(b)
+    if is_monotonically_decreasing(binv)
+        ordered_b = binv ∘ SignFlip() ∘ OrderedBijector() ∘ SignFlip() ∘ b
+    elseif is_monotonically_increasing(binv)
+        ordered_b = binv ∘ OrderedBijector() ∘ b
+    else
+        throw(ArgumentError("ordered transform is currently not supported for $d."))
+    end
+
+    return OrderedDistribution(d, inverse(ordered_b))
+end
+
+bijector(d::OrderedDistribution) = d.bijector
+
+function Distributions._logpdf(d::OrderedDistribution, x::AbstractVector{<:Real})
+    issorted(x) || return -eltype(d)(Inf)
+    return Distributions.logpdf(d.dist, x)
```

Comment:
I would suggest this to avoid any heuristics for determining the return type (`eltype` is related to the eltype of random draws and is unrelated to the type of the return value of `logpdf`):
```suggestion
    lp = Distributions.logpdf(d.dist, x)
    issorted(x) && return lp
    return oftype(lp, -Inf)
```